src/app/product.service.ts

```diff
@@ -0,0 +1,41 @@
+import { map } from 'rxjs/operators';
+import { Injectable } from '@angular/core';
+import { AngularFireDatabase } from '@angular/fire/database';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductService {
+  constructor(private db: AngularFireDatabase) {}
+
+  create(product) {
+    return this.db.list('/products').push(product);
+  }
+
+  getAll() {
+    return this.db
+      .list('/products')
+      .snapshotChanges()
+      .pipe(
+        map((action) =>
+          action.map((a) => {
+            const key = a.payload.key;
```

Comment:
abbreviations are generally not great practice because they dont tell the reader anything about  whats happening. Code needs to be readable for you team more than it needs to work.